src/components/DropDown/DropDown.js

```diff
@@ -119,7 +122,7 @@ const DropDown = React.memo(function DropDown({
             ${textStyle('body2')};
           `}
         >
-          {selectedLabel}
+          {renderLabel || selectedLabel}
```

Comment:
If we move `renderLabel` to be a function, then we could do, earlier:

`const Label = renderLabel`

And here:

```suggestion
          {Label? <Label selectedIndex={selectedIndex} /> : selectedLabel}
```

That way, users will be passing a function that might be a React component, or not.